pkg/kubectl/cmd/apply.go

```diff
@@ -119,7 +121,14 @@ func RunApply(f *cmdutil.Factory, cmd *cobra.Command, out io.Writer, options *Ap
 		}
 
 		if err := info.Get(); err != nil {
-			return cmdutil.AddSourceToErr(fmt.Sprintf("retrieving current configuration of:\n%v\nfrom server for:", info), info.Source, err)
+			if errors.IsNotFound(err) {
+				// Create the resource if it doesn't exist
+				if err := CreateAndRefresh(info); err != nil {
+					return nil
+				}
+			} else {
```

Comment:
Suggest inverting this condition 

    if !errors.IsNotFound(err) {
        ...
    }

so you can return early and avoid the else